pkg/clusteragent/admission/patch/patcher.go

```diff
@@ -102,3 +100,33 @@ func (p *patcher) patchDeployment(req PatchRequest) error {
 	_, err = p.k8sClient.AppsV1().Deployments(req.K8sTarget.Namespace).Patch(context.TODO(), req.K8sTarget.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
 	return err
 }
+
+func enableConfig(deploy *corev1.Deployment, req PatchRequest) error {
+	if deploy.Spec.Template.Labels == nil {
+		deploy.Spec.Template.Labels = make(map[string]string)
+	}
+	deploy.Spec.Template.Labels[common.EnabledLabelKey] = "true"
+	if deploy.Spec.Template.Annotations == nil {
+		deploy.Spec.Template.Annotations = make(map[string]string)
+	}
+	versionAnnotKey := fmt.Sprintf(common.LibVersionAnnotKeyFormat, req.LibConfig.Language)
+	deploy.Spec.Template.Annotations[versionAnnotKey] = req.LibConfig.Version
+	conf, err := json.Marshal(req.LibConfig)
+	if err != nil {
+		return fmt.Errorf("failed to encode library config: %v", err)
+	}
+	configAnnotKey := fmt.Sprintf(common.LibConfigV1AnnotKeyFormat, req.LibConfig.Language)
+	deploy.Spec.Template.Annotations[configAnnotKey] = string(conf)
+	return nil
+}
+
+func disableConfig(deploy *corev1.Deployment, req PatchRequest) {
+	if deploy.Spec.Template.Labels == nil {
+		deploy.Spec.Template.Labels = make(map[string]string)
+	}
+	deploy.Spec.Template.Labels[common.EnabledLabelKey] = "false"
```

Comment:
probably a super nit, but would it make sense to maybe check to see if this label was already set? Maybe worth having some logging for what the previous label was and what we're changing it to (eg. "true" -> "false" or "false" -> "false")?

Reply:
Added a debug log 👍